app/src/main/java/org/oppia/android/app/home/HomeActivity.kt

```diff
@@ -82,12 +89,30 @@ class HomeActivity :
     )
   }
 
-  override fun routeToRecentlyPlayed() {
-    startActivity(
-      RecentlyPlayedActivity.createRecentlyPlayedActivityIntent(
-        this,
-        internalProfileId
-      )
+  override fun routeToRecentlyPlayed(title: String) {
+    val recentlyPlayedActivityParams =
+      RecentlyPlayedActivityParams
+        .newBuilder()
+        .setProfileId(ProfileId.newBuilder().setInternalId(internalProfileId).build())
+        .setActivityTitle(
+          when (title) {
+            getString(R.string.stories_for_you) -> {
```

Comment:
Similar to `getHeader()` function from `PromotedStoryListViewModel.kt` you should use `resourceHandler` to fetch strings here and everywhere else so that you can get translated strings.